src/rotation_generator.jl

```diff
@@ -246,3 +246,36 @@ function Base.show(io::IO, ::MIME"text/plain", X::RotationGenerator)
     io = IOContext(io, :typeinfo => eltype(X))
     Base.print_array(io, X)
 end
+
+"""
+    isrotationgenerator(r)
+
+Check whether `r` is a rotation generator matrix (skew-symmetric matrix).
+"""
+isrotationgenerator
+
+function isrotationgenerator(r::StaticMatrix{N,N,T}) where {N,T<:Real}
+    # This method can be removed, but is a little faster than isrotationgenerator(r::AbstractMatrix{T}).
+    m = SMatrix(r)
+    return m == -m'
+end
+
+function isrotationgenerator(r::AbstractMatrix{T}) where T<:Real
+    indsm, indsn = axes(r)
+    if indsm != indsn
+        return false
+    end
+    for i in indsn, j in i:last(indsn)
```

Comment:
`i:last(indsn)`

Oh that's subtle.  But I think it's correct given the `indsm != indsn` check.

Reply:
Should we replace the line with this?
```julia
    for i in indsm, j in i:last(indsn)
```
In the next line, we have the following script, so it seems okay to assume that both indices have the same axes.
```julia
        if r[i,j] != -r[j,i]
```

Reply:
Oh! now that you mention it, I'd somehow read that as `i:last(indsm)` already...

I think it's fine/good to leave this alone and not deviate from `ishermitian()` more than necessary.